vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SnapshotManager.java

```diff
@@ -0,0 +1,225 @@
+import edu.asu.diging.vspace.core.exception.ImageCouldNotBeStoredException;
+import edu.asu.diging.vspace.core.exception.SnapshotCouldNotBeCreatedException;
+import edu.asu.diging.vspace.core.file.IStorageEngine;
+import edu.asu.diging.vspace.core.model.ISnapshotTask;
+import edu.asu.diging.vspace.core.model.impl.ExhibitionSnapshot;
+import edu.asu.diging.vspace.core.model.impl.SequenceHistory;
+import edu.asu.diging.vspace.core.model.impl.SnapshotTask;
+import edu.asu.diging.vspace.core.model.impl.Space;
+import edu.asu.diging.vspace.core.model.impl.SpaceStatus;
+import edu.asu.diging.vspace.core.services.IRenderingManager;
+import edu.asu.diging.vspace.core.services.ISnapshotManager;
+
+@Service
+@EnableAsync
+@PropertySource("classpath:app.properties")
+public class SnapshotManager  implements  ISnapshotManager {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Value("${page_size}")
+    private int pageSize;
+    
+    @Value("${resources_folder_path}")
+    private String resourcesPath;
+
+    @Autowired
+    @Qualifier("storageEngineDownloads")
+    private IStorageEngine storageEngineDownloads;
+
+    @Autowired
+    private IRenderingManager renderingManager;
+
+    private SequenceHistory sequenceHistory;
+
+    @Autowired
+    private ExhibitionSnapshotRepository exhibitionSnapshotRepository;
+
+    @Autowired
+    private SnapshotTaskRepository snapshotTaskRepository;
+    
+    @Autowired
+    private SpaceRepository spaceRepository;
+    
+    private final String RESOURCES_FOLDER_NAME = "resources";
+    private final String ZIP_FILE_EXTENSION = ".zip";
+
+    /**
+     * Triggers the creation of an exhibition snapshot.
+     *
+     * This method initializes and saves an {@link ExhibitionSnapshot} and a {@link SnapshotTask},
+     *
+     * @return the created {@link ExhibitionSnapshot}
+     * @throws IOException                          if an I/O error occurs during the snapshot creation process
+     * @throws InterruptedException                 if the snapshot creation process is interrupted
+     * @throws SnapshotCouldNotBeCreatedException   if the snapshot could not be created due to any other errors
+     */
+    @Override
+    @Transactional
+    public ExhibitionSnapshot triggerExhibitionSnapshotCreation() throws IOException, InterruptedException, SnapshotCouldNotBeCreatedException {
+        String exhibitionFolderName = getExhibitionFolderName();
+        ExhibitionSnapshot exhibitionSnapshot = new ExhibitionSnapshot();
+        createSnapshotFolder(exhibitionSnapshot, exhibitionFolderName);       
+        SnapshotTask snapshotTask =  createSnapshotTask(exhibitionSnapshot);
+        exhibitionSnapshot.setSnapshotTask(snapshotTask); 
+        exhibitionSnapshot = exhibitionSnapshotRepository.save(exhibitionSnapshot);
+
+        try {
+            snapshotTask = createSnapshot(resourcesPath, exhibitionFolderName, sequenceHistory, exhibitionSnapshot);
```

Comment:
`sequenceHistory` never seems to be initialized?

Reply:
also, `createSnapshot` will not create a new thread as you are calling the method from within this class!